Python/ast_opt.c

```diff
@@ -320,6 +320,28 @@ fold_tuple(expr_ty node, PyArena *arena, _PyASTOptimizeState *state)
     return make_const(node, newval, arena);
 }
 
+static int
+fold_constant_slice(expr_ty node, PyArena *arena, _PyASTOptimizeState *state)
+{
+    PyObject *args[3];
+    expr_ty slices[3] = {
+            node->v.Slice.lower,
+            node->v.Slice.upper,
+            node->v.Slice.step
+    };
+    for (int i = 0; i < 3; i++) {
+        expr_ty slice = slices[i];
+        if (slice == NULL) {
+            args[i] = NULL;
+        } else if (slice->kind == Constant_kind && PyLong_CheckExact(slice->v.Constant.value)) {
+            args[i] = slice->v.Constant.value;
+        } else {
+            return 1;
+        }
+    }
```

Comment:
```suggestion
    if (slice == NULL) {
         PyObject* slice = PySlice_New(NULL, NULL, NULL);
         if (slice == NULL) {
             return NULL;
         }
        return make_const(node, slice, arena);
    }
    
    if (slice->kind != Constant_kind || PyLong_CheckExact(slice->v.Constant.value)) {
            return 1;
     }
     
     PyObject* slice = PySlice_New(node->v.Slice.lower, node->v.Slice.upper, node->v.Slice.step);
     if (slice == NULL) {
         return NULL;
     }
     return make_const(node, PySlice_New(NULL, NULL, NULL), arena);
```

Reply:
I am not sure this would work, since there is no `slice` and we have to individually check the values of `node->v.Slice.<lower/upper/step>` etc.